### ISSUE TITLE
`cclambda` Should Honor `r_convergence`

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#2520]
The `cc` module [forcibly stops users](https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/proc.py#L3149-L3152) from requesting tight convergence out of `cclambda`, so as not to [break tests](https://github.com/psi4/psi4archive/commit/2b2eb6c5a710dde180b1ef7a050012069afeae8d). That's bad.

Deal with #2519 first.

**Comment by lothian** [comment]
I agree that `cclambda` should honor the user's `r_convergence` choice, but the default settings we chose are not to avoid breaking tests but because tighter settings are wholly unnecessary (and often exceedingly difficult to achieve) for transition properties.  The `cc55` test case just revealed the problem on certain hardware.

**Comment by bgpeyton** [comment]
agreed with @lothian -- whoever picks this up, please leave the defaults the same (or else a lot of EOM calculations will suddenly get much harder for no good reason), but the user definitely should be able to crank up the precision if they really want to. may be useful to separate the convergence criteria for the t's and lambdas, if not already done.

**Comment by JonathonMisiewicz** [comment]
> the default settings we chose are not to avoid breaking tests but because tighter settings are wholly unnecessary (and often exceedingly difficult to achieve) for transition properties.

> may be useful to separate the convergence criteria for the t's and lambdas, if not already done.

The running plan is to rename `r_convergence` to `lambda_convergence` or somesuch within `cclambda`, with some logic to auto-set `lambda_convergence` is that is _not_ set but `r_convergence` _is_. `occ` and `dfocc` use a similar trick.

For context, "exceedingly difficult to achieve" means [you're trying to solve ill-conditioned equations](https://github.com/psi4/psi4/issues/2519#issuecomment-1089612589). I'm expecting to be the one who picks this up, and when the time comes, I'll do a quick convergence study to double-check the sensitivity of properties to this parameter.

**Comment by lothian** [comment]
FYI, we did do a brief convergence study, which is how we came up with the 1e-4 criterion.

**Comment by JonathonMisiewicz** [comment]
I figured as much. I'm just paranoid.
